src/wallet/wallet2.cpp

```diff
@@ -6364,7 +6429,26 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
 
           uint64_t i;
           const char *type = "";
-          if (num_found - 1 < recent_outputs_count) // -1 to account for the real one we seeded with
+          if (amount == 0 && has_rct_distribution)
+          {
+            // gamma distribution
+            if (num_found -1 < recent_outputs_count + pre_fork_outputs_count)
```

Comment:
In this case `recent_outputs_count` is known to be 0, right? Why bother adding?

Reply:
Because it matches the ordering so it's a lot clearer.

Reply:
OK